### PR TITLE
feat(json-schema): add option to skip defaults

### PR DESCRIPTION
**Pull request by anshbansal**
- invalid external url were being ingested
- Some folks don't want empty descriptions to be ingested and want them to be ingested/added separately

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/schema/json_schema.py [hunk 1]
```diff
@@ -280,12 +289,14 @@ def _load_one_file(
                 entityUrn=dataset_urn, aspect=models.StatusClass(removed=False)
             ).as_workunit()
 
+            external_url = JsonSchemaTranslator._get_id_from_any_schema(schema_dict)
+            external_url_parsed = urlparse(external_url)
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/schema/json_schema.py:293]: let's wrap this in a try block

Comment (anshbansal) [reply metadata-ingestion/src/datahub/ingestion/source/schema/json_schema.py]: done

File: metadata-ingestion/src/datahub/ingestion/source/schema/json_schema.py [hunk 2]
```diff
@@ -280,18 +290,29 @@ def _load_one_file(
...
+                        "externalUrl", external_url
+                    )
+            except:
+                # Nothing to do in this case
+                pass
+            description = JsonSchemaTranslator._get_description_from_any_schema(
+                schema_dict
+            )
+            if description:
+                dataset_patch_builder.add_custom_property("description", description)
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/schema/json_schema.py:312]: this is for custom properties in the properties tab - use `set_description` instead

File: metadata-ingestion/src/datahub/ingestion/source/schema/json_schema.py [hunk 3]
```diff
@@ -280,18 +290,29 @@ def _load_one_file(
...
+                dataset_simple_name
+            )
+            try:
+                external_url = JsonSchemaTranslator._get_id_from_any_schema(schema_dict)
+                external_url_parsed = urlparse(external_url)
+                if not all([external_url_parsed.scheme, external_url_parsed.netloc]):
+                    external_url = None
+                else:
+                    dataset_patch_builder.add_custom_property(
+                        "externalUrl", external_url
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/schema/json_schema.py:303]: I don't think we actually have support for externalUrl in patch right now

File: metadata-ingestion/src/datahub/ingestion/source/schema/json_schema.py [hunk 4]
```diff
@@ -78,6 +80,14 @@ class JsonSchemaSourceConfig(StatefulIngestionConfigBase, DatasetSourceConfigMix
         default=None,
         description="Use this if URI-s need to be modified during reference resolution. Simple string match - replace capabilities are supported.",
     )
+    inject_default_description: bool = Field(
+        default=True,
+        description="When enabled, injects a default description if one is not present in the schema",
+    )
+    inject_empty_description: bool = Field(
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/schema/json_schema.py:87]: does description being `None` vs "" really change anything? I don't really get the motivation for this flag

File: metadata-ingestion/src/datahub/ingestion/source/schema/json_schema.py [hunk 5]
```diff
@@ -78,6 +80,14 @@ class JsonSchemaSourceConfig(StatefulIngestionConfigBase, DatasetSourceConfigMix
         default=None,
         description="Use this if URI-s need to be modified during reference resolution. Simple string match - replace capabilities are supported.",
     )
+    inject_default_description: bool = Field(
+        default=True,
+        description="When enabled, injects a default description if one is not present in the schema",
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/schema/json_schema.py:85]:
```suggestion
        description="When enabled, includes the default value in the description.",
```

**Pull request closed without merging**